platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultComponentMetadataProcessor.java

```diff
@@ -95,29 +90,19 @@ private static ModuleComponentResolveMetadata realizeMetadata(ModuleComponentRes
 
     private ModuleComponentResolveMetadata forceSerialization(ModuleComponentResolveMetadata metadata) {
         Serializer<ModuleComponentResolveMetadata> serializer = ruleExecutor.getComponentMetadataContextSerializer();
-        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-        byte[] bytes;
-        try {
-            serializer.write(new OutputStreamBackedEncoder(byteArrayOutputStream), metadata);
-            bytes = byteArrayOutputStream.toByteArray();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        } finally {
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            serializer.write(new OutputStreamBackedEncoder(out), metadata);
             try {
-                byteArrayOutputStream.close();
-            } catch (IOException e) {
-                throw new RuntimeException(e);
+                return serializer.read(
+                    new InputStreamBackedDecoder(new ByteArrayInputStream(out.toByteArray())) // Serialize the metadata
+                );
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to deserialize module component metadata", e);
             }
-        }
-        try {
-            ModuleComponentResolveMetadata forceRead = serializer.read(new InputStreamBackedDecoder(new ByteArrayInputStream(bytes)));
-            // TODO: CC cannot enable this assertion because moduleSource is not serialized, so doesn't appear in the deserialized form
-            //assert metadata.equals(forceRead);
```

Comment:
this code now even compiling anymore. Maybe sould extract to issue with dedicated PR until its impossible to restore.